Tests/SwiftDriverTests/SwiftDriverTests.swift

```diff
@@ -313,6 +313,27 @@ final class SwiftDriverTests: XCTestCase {
     XCTAssertEqual(driver6.compilerOutputType, .llvmBitcode)
   }
 
+  func testLtoOutputPath() throws {
+    var driver1 = try Driver(args: ["swiftc", "foo.swift", "-lto=llvm-full",
+                                    "-c", "-target", "x86_64-apple-macosx10.9"])
+    XCTAssertEqual(driver1.compilerOutputType, .llvmBitcode)
+    XCTAssertEqual(driver1.linkerOutputType, nil)
+    let jobs1 = try driver1.planBuild()
+    XCTAssertEqual(jobs1.count, 1)
+    XCTAssertEqual(jobs1[0].outputs.count, 1)
+    XCTAssertEqual(jobs1[0].outputs[0].file.basename, "foo.bc")
+
+    var driver2 = try Driver(args: ["swiftc", "foo.swift", "-lto=llvm-full",
+                                    "-c", "-target", "x86_64-apple-macosx10.9",
+                                    "-o", "foo.o"])
+    XCTAssertEqual(driver2.compilerOutputType, .llvmBitcode)
+    XCTAssertEqual(driver2.linkerOutputType, nil)
+    let jobs2 = try driver2.planBuild()
+    XCTAssertEqual(jobs2.count, 1)
+    XCTAssertEqual(jobs2[0].outputs.count, 1)
+    XCTAssertEqual(jobs2[0].outputs[0].file.basename, "foo.o")
```

Comment:
nit: you can wrap these in `do` blocks to namespace the variables and not need to number everything:
```suggestion
    do {
      var driver = try Driver(args: ["swiftc", "foo.swift", "-lto=llvm-full",
                                      "-c", "-target", "x86_64-apple-macosx10.9"])
      XCTAssertEqual(driver.compilerOutputType, .llvmBitcode)
      XCTAssertEqual(driver.linkerOutputType, nil)
      let jobs = try driver.planBuild()
      XCTAssertEqual(jobs.count, 1)
      XCTAssertEqual(jobs[0].outputs.count, 1)
      XCTAssertEqual(jobs[0].outputs[0].file.basename, "foo.bc")
    }

    do {
      var driver = try Driver(args: ["swiftc", "foo.swift", "-lto=llvm-full",
                                      "-c", "-target", "x86_64-apple-macosx10.9",
                                      "-o", "foo.o"])
      XCTAssertEqual(driver.compilerOutputType, .llvmBitcode)
      XCTAssertEqual(driver.linkerOutputType, nil)
      let jobs = try driver.planBuild()
      XCTAssertEqual(jobs.count, 1)
      XCTAssertEqual(jobs[0].outputs.count, 1)
      XCTAssertEqual(jobs[0].outputs[0].file.basename, "foo.o")
    }
```